src/authn/index.js

```diff
@@ -218,6 +225,19 @@ function setup(app) {
   // be null.
   app.use(passport.session());
 
+  // If the user object was modified in deserializeUser during restore from the
+  // session, then persist the modifications back into the session by updating
+  // the user object in the session via req.login().  The session will detect
+  // its been changed and save itself to the store at the end of the request.
+  app.use((req, res, next) => {
+    if (req.user?.[RESAVE_TO_SESSION]) {
```

Comment:
Wow, didn't know [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) was a thing! 🤓

Reply:
We couldn't use it here until the recent version bump from Node 13 → 14, so it's a nice perk of that! In the same family is also `??`, the "[nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)".